.circleci/config.yml

```diff
@@ -17,6 +17,59 @@
 # Check https://circleci.com/docs/2.0/language-python/ for more details
 #
 version: 2
+
+check_services_status: &check_services_status
+  command: |
+    set -x
+
+    sleep 10  # Give the services enough time to start (or fail)
+
+    for c in as1301 as1303 as1305; do
+      docker-compose exec $c /bin/bash -c \
+        '! systemctl is-failed scion-* || (systemctl status scion-* --no-pager --full && false)'
+    done
+
+check_scion_connections: &check_scion_connections
+  command: |
+    set -x
+
+    # Wait for 3 beacons received from now on
+    docker-compose exec -T as1303 /bin/bash -c \
+      'echo -e "\n\n\n\n\n\n\n\n\n\n" | tee --append /var/log/scion/*;
+       until [ `tail -f /var/log/scion/* | '"awk '/Registered beacons/ {print NR; fflush()}'"' | head -n 1` -ge 3 ]; do sleep 0.1; done;'
```

Comment:
This _seems_ wrong, but I'm not really sure. The `awk | head`  part does not actually count, it prints the line number of first matching "Registered beacons" line. I'm unsure how all of this is interacting with the loop.

One simple idea, if you're using awk anyway, could be this (replacing the entire line, i.e no more `until; do done`):
```
tail -f /var/log/scion/* | awk '/Registered beacons/ {c++; if(c>=3) exit(0)}'
```
Keep a counter `c` as variable in AWK (default initialized with 0) and exit only once this reaches 3.

Reply:
Ah, also, with this approach you can use `tail -n 0 -f ...`, and not do the echo trick just above.